packages/lib-utils/src/k8s/hooks/k8s-watcher.ts

```diff
@@ -0,0 +1,116 @@
+import { CustomError } from '@monorepo/common';
+import * as _ from 'lodash-es';
+import type { Dispatch } from 'redux';
+import * as k8sActions from '../../app/redux/actions/k8s';
+import type { K8sModelCommon } from '../../types/k8s';
+import { getReferenceForModel } from '../k8s-utils';
+import type { GetIDAndDispatch, MakeQuery, Query } from './k8s-watch-types';
+import type { WatchK8sResource } from './watch-resource-types';
+
+export class NoModelError extends CustomError {
+  constructor() {
+    super('Model does not exist');
+  }
+}
+
+export const makeReduxID = (k8sKind: K8sModelCommon, query: Query, cluster?: string) => {
+  let qs = '';
+  if (!_.isEmpty(query)) {
+    qs = `---${JSON.stringify(query)}`;
+  }
+
+  return `${cluster ?? 'local-cluster'}${getReferenceForModel(
+    k8sKind || ({} as K8sModelCommon),
+  )}${qs}`;
+};
+
+export const makeQuery: MakeQuery = (namespace, labelSelector, fieldSelector, name, limit) => {
```

Comment:
I wonder if we could remove the multiple `if`'s and use something like 
```js
export const makeQuery: MakeQuery = (namespace, ...args) => {
  const query: Query = {
    ns: namespace,
    ...args
  };

  return _.pickBy(query, _.identity);
};
```

NOTE: if we expect to pass any falsey values back in the query object this won't work (we'd need to tweak the return filter).

Reply:
Method args are not named... args spread would be an array, not a keyed item.

Reply:
Good point. I'm fine with leaving this.